src/DataSource/Types.hs

```diff
@@ -499,7 +499,8 @@ instance Arbitrary Attachment where
     arbitrary = Attachment
```

Comment:
Is this test code? If so, why is it under `src/` and not `test/`? Can you move it?

Reply:
This is no test code. These are the instances for the `Arbitrary` typeclass. If you don't have any specific and good reason (like the _db_, for example, which I'm still not really sure about either), the instances of the specific type belong where the type is defined.
You can view this as a general arbitrary instance, while we can add more specific ones like we do in the test module(s).
The term you are looking to google is "orphan instances", that may answer some of your questions.

Reply:
Thanks for the pointer. I'm new to Haskell so I'm still trying to understand this.

But if these arbitrary instances are not used anywhere in the source code, and only in the test code, shouldn't they be defined somewhere under the `test` directory?

Reply:
From my perspective, it's easy to maintain the instances if we define in the single module. 

If we start saying "This typeclass is for this module only so we should defined it there"
1. It'll be much harder to maintain the instances since they're defined in separate modules. If we keep everything in `Types.hs` then I know where to look into if I have to update/define new instance.
2. There's definitely going be "gray zone" type classes being introduced. At first we only use that instance for specific module but then we realize we need it for another module as well which will bring confusion. I'm new to Haskell as well so I can't say for sure that which typeclasses are for specific use only or generic. If we maintain them in single file, I don't have to worry about which typeclasses are domain specific and which isn't.

Then there's also orphan instances as well.